### PR TITLE
Load Kubernetes client configuration from usual places

### PR DESCRIPTION
**Pull request by cbandy**
**Checklist:**

 
 - [x] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [x] Have you updated or added documentation for the change, as applicable?
 - [x] Have you tested your changes on all related environments with successful results, as applicable?

**Type of Changes:**

 
 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [x] New feature (non-breaking change which adds functionality)
 - [x] Breaking change (fix or feature that would cause existing functionality to change)

**What is the current behavior? (link to any open issues here)**

OLM scanners expect to be able to redirect Kubernetes API calls by setting environment variables. However, we use in-cluster configuration unless a `--kubeconfig` argument is passed.

**What is the new behavior (if this is a feature change)?**

Rather than a CLI argument, use the configuration loading provided by k8s.io/client-go that checks `KUBECONFIG` environment variable and `~/.kube/config` directory and falls back to in-cluster config, if available.

**Other information**:

This is very similar to #1257 but goes a step further to eliminate the redundant CLI argument.

Issue: [ch7129]

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -74,10 +74,10 @@ deployoperator:
 	cd deploy && ./deploy.sh
 
 runmain: build-postgres-operator
-	postgres-operator --kubeconfig=/etc/kubernetes/admin.conf
+	KUBECONFIG=/etc/kubernetes/admin.conf postgres-operator
 
 runapiserver: build-pgo-apiserver
-	apiserver --kubeconfig=/etc/kubernetes/admin.conf
+	KUBECONFIG=/etc/kubernetes/admin.conf apiserver
```
Comment (jkatz) [inline Makefile]: Hmm...why not put this uptop as `KUBECONFIG ?= /etc/kubernetes/admin.conf`?

Comment (cbandy) [reply Makefile]: I removed these in an amended commit. It seemed like they weren't being used.

File: installers/olm/README.md [hunk 2]
```diff
@@ -1,6 +1,13 @@
 
-This directory contains the files needed to deploy (the Operator) using
-[Operator Lifecycle Manager][OLM].
+This directory contains the files that are used to install [Crunchy PostgreSQL for Kubernetes][hub-listing],
+which uses the PostgreSQL Operator, using [Operator Lifecycle Manager][OLM].
 
-[OLM]: https://github.com/operator-framework/operator-lifecycle-manager
+The integration centers around a [ClusterServiceVersion][olm-csv] [manifest](./postgresoperator.csv.yaml)
+that gets packaged for OperatorHub. Changes there are accepted only if they pass the all the [scorecard][]
```
Comment (abrightwell) [inline installers/olm/README.md]: "~the~ all the"

Comment (cbandy) [reply installers/olm/README.md]: Fixed in amended https://github.com/CrunchyData/postgres-operator/pull/1259/commits/b824c3cd65e8637737413c1ed07df32d47da9117.

**Pull request merged**